### PR TITLE
gRPC stabilization

### PR DESCRIPTION
**Pull request by ggezer**
**Completed:**
- Clients don't send defaults to FileSystem methods
- Selectively disabling DFSIO test based on hadoop client version
- Removed deprecated FSM Rest handler
- Deprecated key/value services
- Guava bumped to latest
- Merge transport and transport-options
- Hardcoded gRPC java plugin
- TODO cleanup

### CODE REVIEW

File: core/client/fs/src/main/java/alluxio/client/file/BaseFileSystem.java [hunk 1]
```diff
@@ -130,13 +130,8 @@ public FileOutStream createFile(AlluxioURI path, CreateFilePOptions options)
...
       // Do not sync before this getStatus, since the UFS file is expected to not exist.
-      GetStatusPOptions gsOptions = FileSystemClientOptions.getGetStatusOptions();
-      gsOptions = gsOptions.toBuilder()
-          .setLoadMetadataType(LoadMetadataPType.NEVER)
-          .setCommonOptions(
-              FileSystemClientOptions.getCommonOptions().toBuilder()
-                  .setSyncIntervalMs(-1))
-          .build();
+      GetStatusPOptions gsOptions =
+          GetStatusPOptions.newBuilder().setLoadMetadataType(LoadMetadataPType.NEVER).build();
```
Comment (calvinjia) [inline core/client/fs/src/main/java/alluxio/client/file/BaseFileSystem.java]: Don't need the sync interval option anymore?

File: core/common/src/main/java/alluxio/security/authentication/DefaultAuthenticationServer.java [hunk 2]
```diff
@@ -56,8 +56,8 @@
   protected final ScheduledExecutorService mScheduler;
 
   /** Interval for clean-up task to fire. */
-  // TODO(gezer) make it configurable.
-  protected final long mCleanupIntervalHour = 1L;
+  protected final long mCleanupIntervalMs =
+      Configuration.getMs(PropertyKey.MASTER_GRPC_SERVER_SHUTDOWN_TIMEOUT);
```
Comment (calvinjia) [inline core/common/src/main/java/alluxio/security/authentication/DefaultAuthenticationServer.java]: This is a very different value, (1 min vs 1 hr), is this intended?

File: bin/alluxio [hunk 3]
```diff
@@ -40,7 +40,6 @@ function printUsage {
   echo -e "  clearCache    \t Clear OS buffer cache of the machine."
   echo -e "  confDocGen    \t Generate configuration docs."
   echo -e "  thriftGen     \t Generate all thrift code."
-  echo -e "  protoGen      \t Generate all protocol buffer code."
```
[the remark this answers is not in the document]
Comment (ggezer) [reply bin/alluxio]: New way is integrated with maven. You don't need to install protoc locally anymore.
In core/transport project installing with "generate" profile will regenerate files. 

> mvn clean install -Pgenerate

File: core/client/fs/src/main/java/alluxio/client/file/BaseFileSystem.java (hunk 1 above)
Comment (ggezer) [reply core/client/fs/src/main/java/alluxio/client/file/BaseFileSystem.java]: -1 is the default value. That's why it's deleted here.
Going forward you don't need to set anything that's default.

File: core/common/src/main/java/alluxio/security/authentication/DefaultAuthenticationServer.java (hunk 2 above)
Comment (ggezer) [reply core/common/src/main/java/alluxio/security/authentication/DefaultAuthenticationServer.java]: Nice catch! Forgot to use the new key.

**Comment by ggezer** [comment]
Jenkins, test this please.

**Pull request closed without merging**